sdcm/nemesis.py

```diff
@@ -7024,3 +7026,40 @@ class SerialRestartOfElectedTopologyCoordinatorNemesis(Nemesis):
 
     def disrupt(self):
         self.disrupt_serial_restart_elected_topology_coordinator()
+
+
+class DisruptionLogger:
+    """A context manager for logging the start, end and duration of a disruption operation, on DB and loader nodes"""
+
+    def __init__(self, nemesis, operation_name, target_node=None, additional_info=None):
+        self.nemesis = nemesis
+        self.operation_name = operation_name
+        self.target_node = target_node or nemesis.target_node
+        self.additional_info = additional_info
+        self.start_time = None
+
+    def __enter__(self):
+        self.start_time = time.time()
+        message = f"Starting disruption: {self.operation_name}"
+        if self.target_node:
+            message += f" on node {self.target_node.name}"
+        if self.additional_info:
+            message += f" - {self.additional_info}"
+
+        for nodes_set in (self.nemesis.cluster, self.nemesis.loaders):
+            nodes_set.log_message(message)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        duration = time.time() - self.start_time
+        status = "failed" if exc_type else "completed"
+        message = f"Disruption {self.operation_name} {status} after {duration:.2f}s"
+        if self.target_node:
+            message += f" on node {self.target_node.name}"
+        if exc_val:
+            message += f" - Error: {exc_val}"
+
+        for nodes_set in (self.nemesis.cluster, self.nemesis.loaders):
```

Comment:
not sure about the loaders, it would in log people rarely look into, and not the actual stress commands logs

Reply:
also I think we should keep a separate log of those as a summery of what SCT was doing.

Reply:
I agree, loaders are not that important. 
I also thought about marking just in 'target node' instead all of them, but might be valuable in other db nodes too.